07/main.py

```diff
@@ -0,0 +1,90 @@
+"""
+Home Work 04
+Dmytro Verovkin
+robot_dreams
+
+
+Створити телефонну книгу, яка матиме наступні команди:
+
+stats: кількість записів
+add: додати запис
+delete <name>: видалити запис за іменем (ключем)
+list: список всіх імен в книзі
+show <name>: детальна інформація по імені
+
+Записи не мають повторюватися, заборонено перезаписувати записи, тільки видаляти і додавати заново.
+"""
+
+
+print("PHONE BOOK")
+print("commands: stats, add <name> <phone>, delete <name>, list, show <name>, exit, help")
+help_text = """
+--------------------------------------------------------------------------------------------------
+PHONEBOOK HELP
+
+Use commands to operate this phonebook:
+stats              - shows number of phonebook entries
+add <name> <phone> - adds <name> and <phone> number to phonebook, if <name> is not already exists
+show <name>        - prints name and phone number for entry <name>
+delete <name>      - deletes entry from phonebook by <name>
+list               - prints all entries of the phonebook
+exit               - exits the program
+help               - shows this help message
+--------------------------------------------------------------------------------------------------
+"""
+
+phone_book = {}
+
+while True:
+    user_input = input("Make your input: ")
+
+    if len(user_input) < 1:             # check if user input is not empty
+        print("Please enter command")
+        continue
+
+    user_input_split = user_input.split()
+
+    match user_input_split[0]:
```

Comment:
Great job! Here is an example of how you can improve your code to use all power of the match-case:
```
phone_book = {}

while True:
    user_input = input("Make your input: ")

    if len(user_input) < 1:             # check if user input is not empty
        print("Please enter command")
        continue

    user_input_split = user_input.split()

    match user_input_split:
        case 'stats', *stats_args:
            print(f"Total {len(phone_book)} entries")
        case 'add', name, phone:
            if name not in phone_book:
                phone_book[name] = phone
                print("Added successfully")
            else:
                print(f"{name} is already exist")
        case 'add', *add_args:
            print("Wrong arguments, please use format: add <name> <phone>")
        case 'delete', name:
            if name in phone_book:
                del phone_book[name]
                print(f"{name} deleted")
            else:
                print(f"{name} is not found")
        case 'delete', *delete_args:
            print("Wrong arguments, please use format: delete <name>")
        case 'list':
            # check if phonebook has entries
            if len(phone_book) > 0:
                for key, value in phone_book.items():
                    print(f"{key} {value}")
            else:
                print("empty")
        case 'show', name:
            if phone_book.get(name) is not None:
                print(f"{name} {phone_book[name]}")
            else:
                print(f"{name} not found")
        case 'show', *show_args:
            print('wrong format')
        case 'exit':
            break
        case 'help':
            print(help_text)
        case _:
            print("Unknown command")